src/main/kotlin/hs/flensburg/marlin/business/api/sensors/boundary/SensorService.kt

```diff
@@ -78,36 +73,31 @@ object SensorService {
             KIO.ok(mapToUnitsWithLocationWithBoxesDTO(rawLocations))
         }
 
-    fun getLocationByIDWithMeasurementsWithinTimespan(
-        locationId: Long,
-        timeRange: String, // "today", "week", "month"
-        timezone: String,
-        ipAddress: String,
-        units: String
-    ): App<Error, LocationWithBoxesDTO?> = KIO.comprehension {
-        val rawLocation = !SensorRepo.fetchLocationByIDWithMeasurementsWithinTimespan(
-            locationId,
-            timeRange,
-            !TimezonesService.getClientTimeZoneFromIPOrQueryParam(timezone, ipAddress),
-            units
-        ).orDie().onNullFail { Error.NotFound }
-        KIO.ok(rawLocation.mapToLocationWithBoxesDTO())
-    }
-
     fun getLocationByIDWithMeasurementsWithinTimespanFAST(
         locationId: Long,
-        timeRange: String, // "today", "week", "month"
+        timeRange: String,
         timezone: String,
         ipAddress: String,
         units: String
     ): App<Error, LocationWithBoxesDTO?> = KIO.comprehension {
-        val rawLocation = !SensorRepo.fetchLocationByIDWithMeasurementsWithinTimespanFAST(
-            locationId,
-            timeRange,
-            !TimezonesService.getClientTimeZoneFromIPOrQueryParam(timezone, ipAddress),
+        val validTimeRange = when (timeRange.lowercase()) {
+            "24h" -> true
+            "48h" -> true
+            "7d" -> true
+            "30d" -> true
+            "90d" -> true
+            "180d" -> true
+            "1y" -> true
+            else -> false
+        }
+        !KIO.failOn(!validTimeRange) { Error.BadRequest }
+
+        val data = !SensorRepo.getLatestMeasurementTimeEnriched(
+            locationId, timeRange,
+            timezone = !TimezonesService.getClientTimeZoneFromIPOrQueryParam(timezone, ipAddress),
             units
         ).orDie().onNullFail { Error.NotFound }
```

Comment:
```suggestion
        SensorRepo.getLatestMeasurementTimeEnriched(
            locationId, timeRange,
            timezone = !TimezonesService.getClientTimeZoneFromIPOrQueryParam(timezone, ipAddress),
            units
        ).orDie().onNullFail { Error.NotFound }.map { it.mapToLocationWithBoxesDTO() }
```
 
Die Funktion `mapToLocationWithBoxesDTO` würde ich eher in `toLocationWithBoxesDTO` umbennen.